### ISSUE TITLE
how to train with the model i have trained before?

### ISSUE THREAD

**Issue by qinhaifangpku** [fchollet/keras#4195]
hi~ all~
now i have train a model with max_iteration = 300
but when i finish the iteration i found that the loss and the accuracy is not as good as i expected.
but one iteration it will cost 3 minites,   now i want to train the model with more iteration, i wonder if i can use the model i have trained on the last 300 iteration, and how i can use it?

is there anyone can give me some advices?  thanks a lot in advance! 

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ 1] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [1 ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ 1] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by superhans** [comment]
Yes you can very easily do this. 
If I'm not mistaken, just load the model you have saved with `keras.models.load_model(filepath)`
And then continue calling the `model.fit` function until you are satisfied. 
I dont believe you have to compile again or anything of that sort. Just `load_model` and `fit`

**Comment by qinhaifangpku** [comment]
@superhans thanks for  your help! 
is that mean i do not have to use the sentens model.compile()? but how it knows that whitch optimizer to use?

**Comment by jf003320018** [comment]
@qinhaifangpku I am agree with @superhans. The optimizer in second training would be the optimizer you used in the first training. If you want to use a different optimizer, you could recompile the model. 
I have not test it, but it should be work.

**Comment by qinhaifangpku** [comment]
@jf003320018 i have tried, but it has an error, it must be compiled. because i use model.load_weights.
maybe i did not save the model, i just save the model weights, howto save a model?
when i load_model it has an error said that i have no model file.

**Comment by jf003320018** [comment]
You can check the links: https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model
There are two ways, one just saves the important things, and the other save the all model.

**Comment by mmishourovsky** [comment]
actually, as we know from issues posted so far loading does not work; it works if one deletes the optimizer section from h5 format, isnt'it? What I am doing then: I load a model, create optimizer, compile model using newly created optimizer and then is able to train. But, is it correct way to do this? it works, but in my case loss does not decrease after 2nd training (probably, its already trained enough) 
Have someone tried this scenario? Or: any ideas how to verify it?
